src/Text/Markdown/SlamDown/Halogen/Component.purs

```diff
@@ -338,85 +378,109 @@ renderSlamDown config (SDS.SlamDownState state) =
-  id ← Fresh.fresh
+  ident ← Fresh.fresh
   let renderedValue = SD.renderValue value
   pure $ HH.li_
     [ HH.input
         [ HP.checked checked
         , HP.type_ HP.InputRadio
-        , HP.id_ id
+        , HP.id_ ident
         , HP.name label
         , HP.value renderedValue
         , HE.onValueChange (HE.input_ (SDQ.ChangeRadioButton label value))
         ]
-    , HH.label [ HP.for id ] [ HH.text renderedValue ]
+    , HH.label [ HP.for ident ] [ HH.text renderedValue ]
     ]
 
 renderCheckBox
-  ∷ ∀ p v
+  ∷ ∀ m v
   . (SD.Value v)
   ⇒ String  -- label
   → v       -- value
-  → FreshRenderer p v Boolean
+  → FreshRenderer v m Boolean
 renderCheckBox label value checked = do
-  id ← Fresh.fresh
+  ident ← Fresh.fresh
   let renderedValue = SD.renderValue value
   pure $ HH.li_
     [ HH.input
         [ HP.checked checked
         , HP.type_ HP.InputCheckbox
-        , HP.id_ id
+        , HP.id_ ident
         , HP.name label
         , HP.value renderedValue
         , HE.onChecked (HE.input (SDQ.ChangeCheckBox label value))
         ]
-    , HH.label [ HP.for id ] [ HH.text renderedValue ]
+    , HH.label [ HP.for ident ] [ HH.text renderedValue ]
     ]
 
 renderDropDown
-  ∷ ∀ p v
+  ∷ ∀ m v
   . (SD.Value v)
   ⇒ String    -- id
   → String    -- label
   → L.List v  -- choices
   → M.Maybe v -- value
-  → H.HTML p (SDQ.SlamDownQuery v)
-renderDropDown id label ls sel =
+  → HTML v m
+renderDropDown ident label ls sel =
   HH.select
-    [ HP.id_ id
+    [ HP.id_ ident
     , HP.name label
     , HE.onSelectedIndexChange (HE.input (SDQ.ChangeDropDown label <<< L.index ls))
     ]
     $ A.fromFoldable $ M.maybe option option' sel <$> ls
   where
-    option ∷ v → H.HTML p (SDQ.SlamDownQuery v)
+    option ∷ v → HTML v m
     option value =
       let renderedValue = SD.renderValue value
       in HH.option [ HP.value renderedValue ] [ HH.text renderedValue ]
 
-    option' ∷ v → v → H.HTML p (SDQ.SlamDownQuery v)
+    option' ∷ v → v → HTML v m
     option' sel' value =
       let renderedValue = SD.renderValue value
       in HH.option [ HP.selected (value == sel'), HP.value renderedValue ] [ HH.text renderedValue ]
 
+data PickerType = DatePicker | TimePicker | DateTimePicker
+textBoxToPicker ∷ ∀ f
+  . BrowserFeatures
+  → SD.TextBox f
+  → Maybe PickerType
+textBoxToPicker features tb = if features.inputTypeSupported (SDIT.textBoxToInputType tb)
+  then Nothing
+  else case tb of
+    SD.Date _ → Just DatePicker
+    SD.Time _ _ → Just TimePicker
+    SD.DateTime _ _ → Just DateTimePicker
+    _ → Nothing
+
+datePickerFormat :: DatePickerF.Format
+datePickerFormat = unsafePartial fromRight $ DatePickerF.fromString "MM:DD:YYYY"
+
+timePickerFormat :: TimePickerF.Format
+timePickerFormat = unsafePartial fromRight $ TimePickerF.fromString "hh:mm a"
+
+dateTimePickerFormat :: DateTimePickerF.Format
+dateTimePickerFormat = unsafePartial fromRight $ DateTimePickerF.fromString "MM:DD:YYYY, hh:mm a"
```

Comment:
Probably don't need the comma here, so `YYYY-MMMM-DD HH:mm` or `YYYY-MMMM-DD HH:mm:ss`, `Precision`-depending.